### PR TITLE
Update Base58 codec and remove Bitcoin support:

### PR DESCRIPTION
**Pull request by cryptobrad**
Reopening the closed PR #3575 sorry fo the problem @scottschurr and @howardhinnant!

### CODE REVIEW

File: src/ripple/protocol/impl/tokens.cpp [hunk 1]
```diff
@@ -246,16 +195,38 @@ decodeBase58(std::string const& s, InverseArray const& inv)
...
-    The type and checksum are are checked
-    and removed from the returned result.
-*/
-template <class InverseArray>
-static std::string
-decodeBase58Token(std::string const& s, TokenType type, InverseArray const& inv)
+std::string
+encodeBase58Token(TokenType type, void const* token, std::size_t size)
+{
+    assert(size != 0);
```
[the remark this answers is not in the document]
Comment (nbougalis) [reply src/ripple/protocol/impl/tokens.cpp:203]: I wonder if we ever do call this with an zero-sized buffer. I can't imagine any case where we do this and it's not either a mistake or an omission and if we were writing this function from scratch, I'd be fine with an `assert` or even with a `throw`.

But given our recent experience with `assert` I'm with @scottschurr here and would prefer to see this `assert` removed.

**Comment by sublimator** [comment]
ohoh, what happened ?

**Comment by scottschurr** [comment]
Oops.  One more mechanical change needed.  The clang-format continuous integration check is failing.

We need to reformat a little bit of the code to make clang-format happy.  You can do that yourself by running clang-format over the changed files.  Or you can cherry-pick the changes that clang-format will make from this commit: https://github.com/scottschurr/rippled/commit/1483ffe434966d64a1278ee695e0e1af98ac22ce

Sorry that I missed that earlier.

**Comment by scottschurr** [comment]
Oh, and since I didn't mention it before, nice job on the commit messages.  Looks great.  :+1:

**Pull request closed without merging**